stl/inc/ranges

```diff
@@ -728,6 +735,482 @@ namespace ranges {
         using all_t = decltype(all(_STD declval<_Rng>()));
     } // namespace views
 
+    // CLASS TEMPLATE ranges::transform_view
+    template <class _Rng, class _Fn> // TRANSITION, LLVM-47414
+    concept _Meow = range<const _Rng>&& regular_invocable<const _Fn&, range_reference_t<const _Rng>>;
+
+    // clang-format off
+    template <input_range _Vw, copy_constructible _Fn>
+        requires view<_Vw> && is_object_v<_Fn>
+            && regular_invocable<_Fn&, range_reference_t<_Vw>>
+            && _Can_reference<invoke_result_t<_Fn&, range_reference_t<_Vw>>>
+    class transform_view : public view_interface<transform_view<_Vw, _Fn>> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+        /* [[no_unique_address]] */ _Semiregular_box<_Fn> _Fun{};
+
+        template <bool _Const>
+        class _Sentinel;
+
+        template <class _Traits, class _Base> // TRANSITION, LWG-3289
+        struct _Iterator_base {};
+        // clang-format off
+        template <class _Traits, class _Base>
+            requires _Has_member_iterator_category<_Traits>
+        struct _Iterator_base<_Traits, _Base> {
+            // clang-format on
+            using iterator_category =
+                conditional_t<is_lvalue_reference_v<invoke_result_t<_Fn&, range_reference_t<_Base>>>,
+                    conditional_t<derived_from<typename _Traits::iterator_category, contiguous_iterator_tag>,
+                        random_access_iterator_tag, typename _Traits::iterator_category>,
+                    input_iterator_tag>;
+        };
+
+        template <bool _Const>
+        class _Iterator : public _Iterator_base<iterator_traits<iterator_t<_Vw>>, _Maybe_const<_Const, _Vw>> {
+        private:
+            template <bool>
+            friend class _Iterator;
+            template <bool>
+            friend class _Sentinel;
+
+            using _Parent_t = _Maybe_const<_Const, transform_view>;
+            using _Base     = _Maybe_const<_Const, _Vw>;
+
+            iterator_t<_Base> _Current{};
+            _Parent_t* _Parent{};
+
+#if _ITERATOR_DEBUG_LEVEL != 0
+            constexpr void _Check_dereference() const noexcept /* terminates */ {

```

Comment:
What "terminates" here - exceptions emitted from `end`?

Reply:
Yes, or from the iterator inequality. All of the callers have conditional `noexcept` that doesn't change to account for these operations in IDL2, terminating closer to the source of the problem seems more informative to users. I'm less of a fan of this idea today: I think it would be an improvement to use a `_NOEXCEPT_IDL0(....)` function-like macro that's defined as `noexcept(__VA_ARGS__)` in IDL0 and an empty string in IDL!=0. That would preserve the strengthening in Release builds without introducing artificial terminates or overcomplicated conditional `noexcept`.